### ISSUE TITLE
Vulkan crashes with VK_ERROR_DEVICE_LOST

### ISSUE THREAD

**Issue by NogNoa** [RPCS3/rpcs3#12579]
### Quick summary

Game crashes midplay. Or the game window locks, unable to be resumed.

### Details

Game: Resistance: Fall of Man™ [NPEA00430]

RPCS3 v0.0.23-14088-698c3415 Alpha

F {RSX [0x28573c4]} SIG: Thread terminated due to fatal error: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\VK\vkutils\sync.cpp:214[:5], in function wait_for_event)
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\VK\vkutils\shared.cpp:107[:24], in function die_with_error)

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/9439334/RPCS3.log.gz)

### System configuration

CPU: Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz | 8 Threads | 31.94 GiB RAM | TSC: 3.598GHz | AVX+ | FMA3
Operating system: Windows, Major: 10, Minor: 0, Build: 19044, Service Pack: none, Compatibility mode: 0
vulkan-compatible GPU: 'NVIDIA GeForce GTX 970' running on driver 516.94.0.0
Firmware version: 4.86

### Other details

Tendency seems level-dependant.
Also seems to be exacerbated by multitasking the gpu, such as watching a video while playing.

**Comment by Darkhost1999** [comment]
Log is empty.
Boot the game.
Create the issue.
Reupload your log.

**Comment by NogNoa** [comment]
Yeah sorry, I thought I didn't overwrote it yet at the time.

For some reason it doesn't let me edit it in.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/9452136/RPCS3.log.gz)

That's a different massage though, but it's crahing all the same.

**Comment by NogNoa** [comment]
Same problem

Game: Resistance: Fall of Man™ [NPEA00430]

RPCS3 v0.0.24-14112-64579eef Alpha | master

F {RSX [0x23d6cdc]} SIG: Thread terminated due to fatal error: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\VK\VKQueryPool.cpp:45[:4], in function poke_query)
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\VK\vkutils\shared.cpp:107[:24], in function die_with_error)

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/9475980/RPCS3.log.gz)

**Comment by Darkhost1999** [comment]
How far do you gotta play to get this error?
Can you trigger the issue every mission?

**Comment by NogNoa** [comment]
I'm sure it's not uniform. But I can't say if there are levels where the fault is impossible. There's a strong coralation between how many time a level crashed on me and how many times I've played it.

The game is made of locations, each having three smaller levels. I think there are locations that are more prone than others, and than there are specific points (rooms and/or battles) where a crash is especially likely. For example now I'm playing the Cheshire location which is very prone, and the third level have a battle where it crashed 3 or 4 times already. But the previous location, Nothingham had barely any crashes at all.

**Comment by NogNoa** [comment]
Game freezes

Game: Resistance: Fall of Man™ [NPEA00430]

RPCS3 v0.0.24-14112-64579eef Alpha | master

F {RSX [0x2384108]} SIG: Thread terminated due to fatal error: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\VK\VKQueryPool.cpp:45[:4], in function poke_query)
(in file C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\rpcs3\Emu\RSX\VK\vkutils\shared.cpp:107[:24], in function die_with_error)

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/9484337/RPCS3.log.gz)

I turned down resolution scaling from 150% to 100%.

**Comment by NogNoa** [comment]
So the final fight in Cheshire caused a crash every single time, until I turned off both the resolution scale and the 60fps patch, which has let the fight conclude on the first try. Turning only the resolution scale down doesn't seem crash-proof, still have to try turning only the 60fps patch off.

**Comment by Darkhost1999** [comment]
I know the error message is an issue:
https://github.com/RPCS3/rpcs3/issues/9133#issuecomment-1229223934
I'm just only at Grimsby and forgot to test the other day.
Issues caused by patches or resolution scaling are not really issues as resolution scaling is a hack and game patches are mods.
The logs I've looked at are not using RSX buffers. So it's not caused by accuracy settings like how my comment on the other issue was.
I'll try to test soon and I'll try to get a debug log output of the error if I am able. In order to make a debug log I need to learn of a consistent way to reproduce the issue. Like a specific trigger every time. It being random makes the test extremely difficult and I'm not That big a fan of this game.

**Comment by NogNoa** [comment]
Yeah, I'm aware hack-releted problems are not issues. That's part of why I tried to isolate them. At first I haven't thought about them, or maybe I tought there were problems before I started using them, I'm not really sure.
The only place I found so far where it was bound to crash was the end of that mission. I actually enjoy the game more than most people seem, but playing the same level again and again is still a bit too much.
Right now the game seem to crash on the final battle of Chshire when there is at least one patch and not when there are none.
I think I'll close the issue.
